pymc3/backends/ndarray.py

```diff
@@ -72,11 +72,16 @@ def load_trace(directory, model=None):
     pm.Multitrace that was saved in the directory
     """
     straces = []
+    assert os.path.isdir(directory), "No such trace directory: " + directory
```

Comment:
This should be a `FileNotFoundError`, instead of an assert.

The difference being that assertions can be turned off in python with the `-O` flag, whereas exceptions are forever. Therefore, proper usage of assertions are for things that:
1) your own code is responsible for, and you therefore "know" to be true, and
2) are serious enough that you would prefer to crash immediately than keep going.

Reply:
Thank you.  I have fixed this and pushed an update (P.S. once everyone's OK with this patch, I will squash it so we have a nice history).
I wasn't sure whether I should use `FileNotFoundError`, since it seemed like that was something that the OS should raise itself.
Please resolve this conversation if the change is approved.